srcCpp/ParameterManager.cxx

```diff
@@ -0,0 +1,233 @@
+    pub->setCommandLineArgument(std::make_pair("-pub",""));
+    pub->setDescription("Set test to be a publisher");
+    pub->setType(T_BOOL);
+    pub->setExtraArgument(NO);
+    parameterList["pub"] = AnyParameter(pub);
+
+
+    ParameterVector<std::string> * peer = new ParameterVector<std::string>();
+    peer->setCommandLineArgument(std::make_pair("-peer","<address>"));
+    peer->setDescription("Adds a peer to the peer host address list.\nThis argument may be repeated to indicate multiple peers");
+    peer->setType(T_VECTOR_STR);
+    peer->setExtraArgument(YES);
+    parameterList["peer"] = AnyParameter(peer);
+
+    std::vector<unsigned long long> _scanDataLenSizes;
+    _scanDataLenSizes.push_back(32);
+    _scanDataLenSizes.push_back(64);
+    _scanDataLenSizes.push_back(128);
+    _scanDataLenSizes.push_back(256);
+    _scanDataLenSizes.push_back(512);
+    _scanDataLenSizes.push_back(1024);
+    _scanDataLenSizes.push_back(2048);
+    _scanDataLenSizes.push_back(4096);
+    _scanDataLenSizes.push_back(8192);
+    _scanDataLenSizes.push_back(16384);
+    _scanDataLenSizes.push_back(32768);
+    _scanDataLenSizes.push_back(63000);
+    ParameterVector<unsigned long long> * scan = new ParameterVector<unsigned long long>(_scanDataLenSizes);
+    scan->setCommandLineArgument(std::make_pair("-scan","<size1>:<size2>:...:<sizeN>"));
+    scan->setDescription("Run test in scan mode, traversing\na range of sample data sizes from\n[32,63000] or [63001,2147483128] bytes,\nin the case that you are using large data or not.\nThe list of sizes is optional.\nDefault values are '32:64:128:256:512:1024:2048:4096:8192:16384:32768:63000'\nDefault: Not set\n");
+    scan->setType(T_VECTOR_NUMERIC);
+    scan->setExtraArgument(OPTIONAL);
+    scan->setRange(32, 2147483128);
+    scan->setParseMethod(SPLIT);
+    parameterList["scan"] = AnyParameter(scan);
+
+
+    Parameter<unsigned long long> *unbounded = new Parameter<unsigned long long>(63000);
+    unbounded->setCommandLineArgument(std::make_pair("-unbounded","<allocation_threshold>"));
+    unbounded->setDescription("Use unbounded Sequences\n<allocation_threshold> is optional, default 63000 Bytes.");
+    unbounded->setType(T_NUMERIC);
+    unbounded->setExtraArgument(OPTIONAL);
+    unbounded->setRange(0, 2147483128);
+    parameterList["unbounded"] = AnyParameter(unbounded);
+
+}
+
+
+// Parse the command line parameters and set the value
+bool ParameterManager::parse(int argc, char *argv[])
+{
+    unsigned long long var;
+    bool success = true;
+    // Copy all arguments into a container of strings
+    std::vector<std::string> allArgs(argv, argv + argc);
+
+    std::map<std::string, AnyParameter>::iterator it;
+    for (unsigned int i = 1; i < allArgs.size(); i++) {
+        for (it = parameterList.begin(); it != parameterList.end(); it++) {
+            if (allArgs[i] == it->second.get()->getCommandLineArgument().first) { // TODO check for small string compare
```

Comment:
Maybe you can skip a 'for loop' with: 
AnyParameter anyParam = parameterList[allArgs[i].c_str() + sizeof(char)]

The  `+ sizeof(char) `  is for skip the first character on the string: from `-pub` to  `pub` for example

Reply:
After discussing with @alfonso-soto, this suggestion is not covering all the use case. Thank you